specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/stable/2020-05-15/timeseriesinsights.json

```diff
@@ -0,0 +1,2549 @@
+        }
+      }
+    },
+    "AccessPolicyMutableProperties": {
+      "description": "An object that represents a set of mutable access policy resource properties.",
+      "type": "object",
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "An description of the access policy."
+        },
+        "roles": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "A role defining the data plane operations that a principal can perform on a Time Series Insights client.",
+            "enum": [
+              "Reader",
+              "Contributor"
+            ],
+            "x-ms-enum": {
+              "name": "AccessPolicyRole",
+              "modelAsString": true
+            }
+          },
+          "description": "The list of roles the principal is assigned on the environment."
+        }
+      }
+    },
+    "CloudError": {
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CloudErrorBody",
+          "description": "Describes a particular API error with an error code and a message."
+        }
+      },
+      "description": "Contains information about an API error.",
+      "x-ms-external": true
+    },
+    "CloudErrorBody": {
+      "type": "object",
+      "description": "Describes a particular API error with an error code and a message.",
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "An error code that describes the error condition more precisely than an HTTP status code. Can be used to programmatically handle specific error cases."
+        },
+        "message": {
+          "type": "string",
+          "description": "A message that describes the error in detail and provides debugging information."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the particular error (for example, the name of the property in error)."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody",
+            "description": "Describes a particular API error with an error code and a message."
+          },
+          "description": "Contains nested errors that are related to this error."
+        }
+      },
+      "x-ms-external": true
+    }
+  },
+  "parameters": {
+    "SubscriptionIdParameter": {
+      "name": "subscriptionId",
+      "in": "path",
+      "description": "Azure Subscription ID.",
+      "required": true,
+      "type": "string"
+    },
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "Version of the API to be used with the client request."
+    },
+    "ExpandParameter": {
+      "name": "$expand",
```

Comment:
Should be "expand" now, instead of "$expand", right?

Reply:
ARM tries to follow OData (a tiny subset, really), so this should actually be `$expand`.

---
In reply to: [398956944](https://github.com/Azure/azure-rest-api-specs/pull/8791#discussion_r398956944) [](ancestors = 398956944)

Reply:
Agreed - and the parameter used to be $expand just to meet OData standards as you say. However, ARM broke that underneath us some time last spring - calls  that used the "$expand" started failing with an error saying something like "$" is a reserved character. Since analysis showed this query parameter was only really used by our Ibiza extension, we switched to "expand" to fix the issue.
I had intended to raise this issue with ARM, but it fell off my plate.

Reply:
I don't believe that ARM was the cause of that bug because there were millions of successful requests with $expand in the last hour alone.

---
In reply to: [401840532](https://github.com/Azure/azure-rest-api-specs/pull/8791#discussion_r401840532) [](ancestors = 401840532)